#!/usr/bin/env python

PACKAGE = "lidar_object_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# ROI cfg parameters
gen.add("xMinROI", double_t, 0, "xMinROI", -10.0, -30.0, 0.0)
gen.add("xMaxROI", double_t, 0, "xMaxROI", 30.0, 0.0, 30.0)
gen.add("yMinROI", double_t, 0, "yMinROI", -13.0, -30.0, 0.0)
gen.add("yMaxROI", double_t, 0, "yMaxROI", 13.0, 0.0, 30.0)
gen.add("zMinROI", double_t, 0, "zMinROI", -0.2, -2.0, 0.0) 
gen.add("zMaxROI", double_t, 0, "zMaxROI", 6.0, 0.0, 10.0)

# DBScan cfg parameters
gen.add("minPoints", int_t, 0, "minPoints", 5, 0, 100)
gen.add("epsilon", double_t, 0, "epsilon", 1.0, 0.0, 10.0)
gen.add("minClusterSize", int_t, 0, "minClusterSize", 5, 0, 100)
gen.add("maxClusterSize", int_t, 0, "maxClusterSize", 3000, 10, 10000)

# BoundingBox cfg parameters
gen.add("xMinBoundingBox", double_t, 0, "xMinBoundingBox", 0.01, 0.0, 1.0);
gen.add("xMaxBoundingBox", double_t, 0, "xMaxBoundingBox", 2.0, 1.0, 10.0);
gen.add("yMinBoundingBox", double_t, 0, "yMinBoundingBox", 0.1, 0.0, 1.0);
gen.add("yMaxBoundingBox", double_t, 0, "yMaxBoundingBox", 3.5, 1.0, 10.0);
gen.add("zMinBoundingBox", double_t, 0, "zMinBoundingBox", 0.0, 0.0, 1.0);
gen.add("zMaxBoundingBox", double_t, 0, "zMaxBoundingBox", 2.0, 0.0, 10.0);

# VoxelGrid parameter
gen.add("leafSize", double_t, 0, "leafSizeValue", 0.1, 0, 1)

# Plane parameter
gen.add("maxIterations", int_t, 0, "maxIterations", 10, 0, 10000) # 100
gen.add("distanceThreshold", double_t, 0, "distanceThreshold", 0.2, 0, 100) # 0.3

exit(gen.generate(PACKAGE, "lidar_object_detection_dynamic_obstacle_node", "objectDetectorDynamicObstacle"))